### ISSUE TITLE
[Bug]: Incorrect return type generated for Model::findOrNew()

### ISSUE THREAD

**Issue by dmason30** [laravel-idea/plugin#397]
### Bug description

Running the Generator for eloquent generates the following for `findOrNew` type hinting it as a collection:

```php
* @method static _IH_User_C|User[] findOrNew($id, array $columns = ['*'])
```

The problem is that this method will actually return a single model instance. The correctly generated phpdoc should be:

```php
* @method static User findOrNew($id, array $columns = ['*'])
```

### Plugin version

4.4.0.212

### Operating system

MacOS

**Comment by adelf** [comment]
Hi, Daniel. You posted it right when I was working on Eloquent helper code) Will be fixed. Thank you.

BTW, this method can return 1 model or a collection of models. So, the phpDoc will be 

```php
* @method static User|_IH_User_C|User[] findOrNew($id, array $columns = ['*'])
```

**Comment by dmason30** [comment]
@adelf Thanks, realising it can return either is scary, back to code to stop using that method now haha. ❤️

**Comment by adelf** [comment]
Agree. `OrNew` says that it can create 1 model... but at the same time, it can return 1 or several models...

**Comment by adelf** [comment]
It should be fixed by the 4.4.2 version.
